tracer/src/Datadog.Trace/Logging/Internal/DatadogLogging.cs

```diff
@@ -218,5 +221,40 @@ internal static string GetLogDirectory()
 
             return logDirectory;
         }
+
+        internal static void CleanLogFiles(string logsDirectory)
+        {
+            var logDaysLimit = EnvironmentHelpers.GetEnvironmentVariable(ConfigurationKeys.MaxLogFilesDays);
+            if (int.TryParse(logDaysLimit, out var days) && (days > 0 && days < int.MaxValue))
+            {
+                var date = DateTime.Now.AddDays(-1 * days);
+                var logFormats = new[]
+                  {
+                    "dotnet-tracer-managed-*-.log",
+                    "dotnet-tracer-native-*-.log",
+                    "dotnet-native-loader-*.log",
+                    "DD-DotNet-Profiler-Native-*.log"
+                  };
+
+                foreach (var logFormat in logFormats)
+                {
+                    foreach (var logFile in Directory.EnumerateFiles(logsDirectory, logFormat))
+                    {
+                        try
```

Comment:
It's possible for Directory.EnumeratFiles to throw an IOException. If the plan is to error out on the first IO exception, then I suggest moving the try/catch outside of the two foreach loops